pkg/repo/generic/goget.go

```diff
@@ -0,0 +1,162 @@
+package generics
```

Comment:
do we need this `generics` package for a specific reason? if not, I think it would be a better idea to put `func NewGenericFetcher` directly in the `repo` package

Reply:
My idea was to have this generic one and then few specific like github which is unusable right now :D then in repo package i would like to have factory which based on module name tries specific.

But i think neither of this is a dis for moving it one level up